pkg/component/runtime/command.go

```diff
@@ -195,7 +195,7 @@ func (c *commandRuntime) Run(ctx context.Context, comm Communicator) error {
 				// first check-in
 				sendExpected = true
 			}
-			c.lastCheckin = time.Now().UTC()
+			c.lastCheckin = time.Now()
```

Comment:
Interesting, didn't know that the `.UTC()` would prevent a monotonic clock usage in golang.

I think it would be good to add a comment over this line explaining the usage of `time.Now()` and how that ensures monotonic clock. Maybe linking to golang documentation for that would be nice to have.

Reply:
> Because t.In, t.Local, and t.UTC are used for their effect on the interpretation of the wall time, they also strip any monotonic clock reading from their results. The canonical way to strip a monotonic clock reading is to use t = t.Round(0).

They document this in the package level comment of https://pkg.go.dev/time, but not actually on https://pkg.go.dev/time#Time.UTC

Agree with including this as an explicit comment.

It would also be great to be able to write a test for this, but I can't think of a way that doesn't involve manipulating the clock on the machine running the tests.

Reply:
Added comment.

https://github.com/elastic/elastic-agent/issues/5285 to track creating a static analysis to catch

Reply:
Thanks for adding the comment.